src/test.core.c

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <stdio.h>
```

Comment:
Failing to find a good reference (some people's personal opinions on SO and the like), but I would always put standard library #includes _last_.

Reason? If any of the project headers need stdio, but don't #include it, then this is masking another consume not doing the initial #include for stdio.h.

Reply:
That's a really compelling argument, I hadn't considered that. 

In general our coding style has tried to keep C++ and Python as similar as possible, and Python's PEP8 says system headers (imports) come first.

Handily, this also somewhat matches cpplint (which apparently cannot be configured otherwise), which is based on [Google's style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes), which we decided early on to follow (for the most part).  I.e.

> Include headers in the following order: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.

Reply:
Yep, agree it's inverted to Python, as I just got it confused on another review I'm doing today, and needed to go and look up the PEP! :P

Reply:
Another way to ask the question is two-fold:
* how many system headers do OpenAssetIO headers depend upon? (if none, then moot point)
* how annoyed do you think users would be if consumer code failed to compile because an OpenAssetIO header is erroring with 'can't find <xyz> definition'? - of course, these are developers as users, so they'd solve it, but you might get a bug report.

Reply:
I think it may be out of scope to change our style guide in this PR, though I'll raise it with the team.  The main argument in this specific case, i.e. ensuring no implicit dependencies between headers, would be solved by your other excellent suggestion in https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/1#discussion_r1013856218 (to have a separate compilation unit per header).